### DISCUSSION TITLE
Code inside an event cannot access Flask's app context

### DISCUSSION THREAD

**Discussion by allansifuna** [miguelgrinberg/python-socketio#1130] [Q&A] [answered]
I have a flask application as a client application and am using socketio.Client to connect it to the server. I have an event that data is passed into then the event writes the data into the database. I am using Flask_SQLAlchemy for the models. I am facing a major issue in that code inside the event cannot write or query the database using the models since its working outside the app context. I have tried pushing the app context using `current_app` from flask but that too throws the same Working outside of application context error. This used to work fine a while ago but with updates to Flask_SQLAlchemy required that access to models should have an application context. Has anyone faced the same issue or have any ideas of making the socketio events have access to the models?

**Comment by miguelgrinberg** [comment, marked as answer]
Have you tried using the actual application instance to push the app context instead of `current_app`? That's the correct way to do it.

**Reply by allansifuna** [reply]
Yeah that had worked but i was unsure if it was a good convention to do it that way since the project uses blueprints.

**Reply by miguelgrinberg** [reply]
This is a common misunderstanding a lot of Flask developers have. The `current_app` variable cannot do magic, it has no way to know what is your application, it only knows what the application is when an app context was already pushed.
